samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -48,32 +54,33 @@
   // with locality. Since host-affinity is not yet implemented, this can be fixed as part of SAMZA-1197
   private static final int METADATA_CACHE_TTL_MS = 5000;
 
-
   private final ZkUtils zkUtils;
   private final String processorId;
   private final ZkController zkController;
 
   private final Config config;
   private final ZkBarrierForVersionUpgrade barrier;
+  private final ZkJobCoordinatorMetrics metrics;
+  private final Map<String, MetricsReporter> reporters;
 
   private StreamMetadataCache streamMetadataCache = null;
   private ScheduleAfterDebounceTime debounceTimer = null;
   private JobCoordinatorListener coordinatorListener = null;
   private JobModel newJobModel;
-
   private int debounceTimeMs;
 
-  public ZkJobCoordinator(Config config) {
+  public ZkJobCoordinator(Config config, MetricsRegistry metricsRegistry) {
     this.config = config;
     ZkConfig zkConfig = new ZkConfig(config);
-    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getGlobalAppId());
+    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getAppId());
```

Comment:
Is the change from getGlobalAppId and getAppId intentional? If not, please change it back as it will cause a severe bug for zk based deployments (as seen during 0.13.0 release testing and fixed in SAMZA-1326: StreamProcessor should use ApplicationConfig.getGlobalAppId()).

It is not very obvious what a "globalAppId" is. sorry about that. Please correct it.